src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/out-string/out-string.cs

```diff
@@ -45,6 +45,25 @@ public int Width
         }
 
         private Nullable<int> _width = null;
+
+        /// <summary>
+        /// False to add a newline to the end of the output string, true if not.
+        /// </summary>
+        [Parameter]
+        public SwitchParameter NoNewline
```

Comment:
Please put `NoNewLine` and `Sream` parameters in different parameter sets as discussed in the Issue.

Reply:
Done. I have named the ParameterSets as _NonStreamFormatting_ and _StreamFormatting_ for want of a better name. Do suggest one if this is bad :)
I've also fixed-up the commit instead of adding a new one to keep it clean. Let me know if that is an inconvenience.